Duplicati/Library/Main/Operation/Backup/StreamBlockSplitter.cs

```diff
@@ -103,13 +103,10 @@ public static Task Run(Options options, BackupDatabase database, ITaskReader tas
                             var lastread = 0;
                             var buf = new byte[blocksize];
                             var lastupdate = DateTime.Now;
-                            var firstBlock = true;
 
                             // Core processing loop, read blocks of data and hash individually
-                            while (((lastread = await stream.ForceStreamReadAsync(buf, blocksize)) != 0) || firstBlock)
+                            while (((lastread = await stream.ForceStreamReadAsync(buf, blocksize)) != 0))

```

Comment:
I only found one query that was failing because of this - the crazy join you commented on above. I did a bit of manual testing and ran all unit tests with positive results. 

I'm still not 100% convinced that nothing blows up with this change in corner cases, but we'll need to test this branch thoroughly before making it available to the general public anyway.